### DISCUSSION TITLE
Add graceful error-handling when a non-unique value is used for a SlugRelatedField

### DISCUSSION THREAD

**Discussion by allynt** [encode/django-rest-framework#8962] [Ideas & Suggestions]
At the moment, if you specify a field that is not unique as the "slug_field" of a `SlugRelatedField` and the value you pass into the serializer matches multiple models then DRF will crash.  This is in contrast to what happens when the value you pass matches zero models; then DRF handles the error gracefully.

I know that the [documentation](https://www.django-rest-framework.org/api-guide/relations/#slugrelatedfield) recommends only using unique fields as the "slug_field".  

But it would be simple to add one more case to the following code:

https://github.com/encode/django-rest-framework/blob/1ce0853ac51635526dc1a285e6b83c9848002f0e/rest_framework/relations.py#L463-L466

Something like

```Python
except MultipleObjectsReturned:
    self.fail('multiple_objects', slug_name=self.slug_field, value=smart_str(data))
```

Obviously there would have to be a new "multiple_objects" entry added to the `default_error_messages` dictionary.

This would return a properly formatted 400 response when used correctly in a view instead of the current situation which raises a server error.

**Comment by auvipy** [comment]
that check would be nice to have. btw, can this be handled in django level?

**Reply by allynt** [reply]
@auvipy what do you mean by "django level"?
